src/overlays/actors/ovl_Dm_Char09/z_dm_char09.c

```diff
@@ -32,28 +30,197 @@ const ActorInit Dm_Char09_InitVars = {
+    Math_Vec3f_Diff(&sp58, &thisx->world.pos, &thisx->velocity);
+    sp54 = Math3D_Vec3fMagnitude(&thisx->velocity);
+    if ((sp54 < (this->speed * 8.0f)) && (this->speed > 2.0f)) {
+        phi_fv0 = ((this->speed - 2.0f) * 0.1f) + 2.0f;
+        phi_fa0 = this->speed * 0.03f;
+    } else {
+        phi_fv0 = this->speed;
+        phi_fa0 = this->speed * 0.16f;
+    }
+    Math_StepToF(&thisx->speedXZ, phi_fv0, phi_fa0);
+    if ((thisx->speedXZ + 0.05f) < sp54) {
+        Math_Vec3f_Scale(&thisx->velocity, thisx->speedXZ / sp54);
+        thisx->world.pos.x += thisx->velocity.x;
+        thisx->world.pos.y += thisx->velocity.y;
+        thisx->world.pos.z += thisx->velocity.z;
+    } else {
+        this->unk_21C += this->unk_220;
+        thisx->speedXZ *= 0.4f;
+        phi_a1 = true;
+        if (((this->unk_21C >= this->unk_218) && (this->unk_220 > 0)) ||
+            ((this->unk_21C <= 0) && (this->unk_220 < 0))) {
+            temp_v1 = this->unk_224 + this->unk_218;
+            if (((this->unk_224->x == temp_v1->x) && (this->unk_224->y == temp_v1->y)) &&
+                (this->unk_224->z == temp_v1->z)) {
+                this->unk_21C = 0;
+                this->unk_220 = 1;
+            } else {
+                phi_a1 = false;
+                this->actionFunc = func_80AB2268;
+            }
+        }
+        if (phi_a1) {
+            func_80AB1FA0(this, this->unk_21C);
+        }
+    }
+    Math_SmoothStepToS(&thisx->world.rot.y, Math_Vec3f_Yaw(&thisx->world.pos, &sp40) + 0x7FFF, 1, 0x7D0, 0);
+    thisx->shape.rot.y = thisx->world.rot.y;
+}
+
+void DmChar09_DoNothing(DmChar09* this, PlayState* play) {
+}
+
+void func_80AB2268(DmChar09* this, PlayState* play) {
+    Path* path;
+    s32 pad;
+    s32 i;
+    s32 actionIndex;
+    s32 action = 0;
+    s32 pathnum;
+    u8 csIndex = 0;
+
+    if (!DMCHAR09_GET_F(&this->actor)) {
+        if (play->csCtx.currentCsIndex == 1) {
+            csIndex = 1;
+        }
+    } else if (play->csCtx.currentCsIndex == 0) {
+        csIndex = 1;
+    }
+
+    if (Cutscene_CheckActorAction(play, 0x1F7) && (csIndex)) {
+        actionIndex = Cutscene_GetActorActionIndex(play, 0x1F7);
+        if (this->unk_22F != play->csCtx.actorActions[actionIndex]->action) {
+            this->unk_22F = play->csCtx.actorActions[actionIndex]->action;
+            switch (play->csCtx.actorActions[actionIndex]->action) {
+                case 2:
+                    action = 0;
+                    break;
+                case 3:
+                    action = 1;
+                    break;
+                case 4:
+                    action = 2;
+                    break;
+                case 5:
+                    action = 3;
+                    break;
+            }
+
+            if (play->csCtx.actorActions[actionIndex]->action >= 2) {
+                pathnum = DMCHAR09_GET_PATH(&this->actor);
+                path = &play->setupPathList[pathnum];
+
+                for (i = 0; i < action; i++) {
```

Comment:
Based on this, action doesn't really seem to be an `action` more like `limit` or `max`.